system/Queue/Handlers/DatabaseHandler.php

```diff
@@ -0,0 +1,180 @@
+<?php namespace CodeIgniter\Queue\Handlers;
+
+/**
+ * Queue handler for database.
+ */
+class DatabaseHandler implements QueueHandlerInterface
+{
+	const STATUS_WAITING   = 10;
+	const STATUS_EXECUTING = 20;
+	const STATUS_DONE      = 30;
+	const STATUS_FAILED    = 40;
+	protected $group_config;
+	protected $config;
+	protected $db;
+
+	/**
+	 * constructor.
+	 *
+	 * @param  array $group_config
+	 * @param  \Codeigniter\Config\Queue $config
+	 */
+	public function __construct($group_config, \Codeigniter\Config\Queue $config)
+	{
+		$this->group_config = $group_config;
+		$this->config       = clone $config;
```

Comment:
I'm probably missing something here, but why clone the config? Do they get modified somewhere?